neurom/features/population.py

```diff
@@ -77,11 +75,7 @@ def sholl_frequency(
     neurite_filter = is_type(neurite_type)
 
     if bins is None:
-        section_iterator = (
-            partial(iter_sections, section_filter=neurite_filter)
-            if use_subtrees
-            else partial(iter_sections, neurite_filter=neurite_filter)
-        )
+        section_iterator = partial(iter_sections, neurite_filter=neurite_filter)
```

Comment:
This doesn't look right. In the subtree case the filter does not propagate to the section level if the section_filter is not specified. Right?

Reply:
There is something weird here indeed.
This `section_iterator` is only used to compute the bins if they are not provided by the user. So if we filter the sections here, the bins might not be consistent with the result of `neurom.features.morphology.sholl_crossings` that is called later because this function does not filter the sections.
So for now I prefer making the filters consistent and later we can add section filter if it is relevant.

Reply:
Both `sholl_crossings` and `sholl_frequency` should work on the sections of the subtype in the case of subtrees. By filtering only by neurite type this is lost.

Reply:
Indeed, good catch.